design-system-docs/src/_components/shared/component_example.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "htmlbeautifier"
+
+module Shared
+  class ComponentExample < ViewComponent::Base
+    include Bridgetown::ViewComponentHelpers
+
+    def initialize(slug, iframe: false)
+      super
+
+      @slug = slug
+      @iframe = iframe
+      @site = Bridgetown::Current.site
+    end
+
+    def render?
+      find_example.present?
+    end
+
+    def example
+      example_resource = find_example
+      # Make sure that rendered output is processed
+      example_resource.transformer.process!
+      example_resource
+    end
+
+    def find_example
+      @site.collections.component_examples.resources.find do |resource|
+        resource.data.slug == @slug
+      end
+    end
+
+    def iframe?
+      @iframe.present?
+    end
+
+    def highlighted_source
+      # @FIXME: There's probably some helper method to do this for us
+      lexer = Rouge::Lexers::ERB.new
+      formatter.format(lexer.lex(example.untransformed_content))
```

Comment:
Learning question - is `untransformed_content` something that comes directly from the Bridgetown gem? I've tried looking for it but could only find this: https://www.rubydoc.info/gems/bridgetown-core/Bridgetown/Resource/Base#untransformed_content-instance_method

Reply:
Yep that's the exact method! Each resource in bridgetown terms has a few methods we can call. This one as the name suggests is for the untransformed content i.e. the source code for the example.